### PR TITLE
Debuggable lattices

### PR DESCRIPTION
**Pull request by regexident**
Some of the lattice types don't impl `std::fmt::Debug`, making it tedious to debug a program's state.

### CODE REVIEW

File: ascent_base/src/lattice/constant_propagation.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@ use std::cmp::Ordering;
 use super::{BoundedLattice, Lattice};
 
 /// A flat `Lattice`: `Bottom` <= everything <= `Top`, and `Constant(x) == Constant(y)` iff `x == y`
-#[derive(PartialEq, Eq, Clone, Copy, Debug, Hash)]
+#[derive(PartialEq, Eq, Clone, Copy, Hash, Debug)]
```
Comment (s-arash) [inline ascent_base/src/lattice/constant_propagation.rs:6]: Please undo this change.

File: ascent_base/src/lattice/ord_lattice.rs [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
 
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct OrdLattice<T>(pub T);
 
+impl<T: Debug> Debug for OrdLattice<T> {
+   fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result { self.0.fmt(f) }
+}
+
+impl<T: Debug> Display for OrdLattice<T> {
```
Comment (s-arash) [inline ascent_base/src/lattice/ord_lattice.rs:12]: I don't think this impl is appropriate.

Comment (regexident) [reply ascent_base/src/lattice/ord_lattice.rs]: Fixed.

File: ascent_base/src/lattice/constant_propagation.rs (hunk 1 above)
Comment (regexident) [reply ascent_base/src/lattice/constant_propagation.rs]: Done.

**Review by s-arash — APPROVED** [review]
